Projects/Server/Items/Container.cs

```diff
@@ -1533,7 +1533,7 @@ private static Item RecurseFindItemByType(Item current, Type[] types, bool recur
         var container = queue.Dequeue();
         foreach (var item in container.Items)
         {
-          if (item is T typedItem && (predicate == null || predicate(typedItem)))
+          if (item is T typedItem && predicate?.Invoke(typedItem) != null)

```

Comment:
One of the checks is not null and another is not false.
Should be the same? I think is not false?

Reply:
Huh, I'm not sure how that happened. I just copy-pasted the same line across. I'll have it fixed in the next commit. I just wrapped up on your other suggestion with the Queue Pool.